@core/cms/src/external-auth-integration/auth-providers/basic-auth/@types/index.ts

```diff
@@ -0,0 +1,61 @@
+export type BasicAuthString = `basic-${string}`;
+
+/**
+ * Represents the client credential for Basic HTTP Authentication.
+ * @export
+ * @interface ICredential
+ * @property {string} id - The client identifier.
+ * @property {string} secret - The client secret.
+ */
+export interface BasicAuthCredential {
+  username: string;
+  password: string;
+}
+
+/**
+ * Defines the configuration for an OAuth2 provider's endpoints.
+ * @export
+ * @interface IProviderConfiguration
+ * @property {string} authorizeUrl - The URL for the OAuth2 authorization endpoint.
+ * @property {string} tokenUrl - The URL for the OAuth2 token endpoint.
+ * @property {string} callbackUri - The URI for the OAuth2 callback after authentication.·
+ * @property {string} [verifyTokenUrl] - The URL for the OAuth2 token verification endpoint.
+ * @property {string} refreshTokenUrl - The URL for the OAuth2 token refresh endpoint.
+ * @property {number} [tokenRefreshBuffer] - Time in seconds before expiration to refresh the token.
+ * @property {{ [key: string]: string }} [callbackUriParams] - Optional additional parameters for the callback URI.
+ */
+export interface BasicAuthProvider {
+  authenticateUrl: string;
+  verifyUrl?: string;
+}
+
+/**
+ * Represents the full configuration for OAuth2 authentication.
+ * @export
+ * @interface IOAuth2Config
+ * @property {string[]} scope - The list of OAuth2 scopes requested.
+ * @property {ICredentials} credentials - The client credentials.
+ * @property {IProviderConfiguration} provider - The OAuth2 provider's endpoint configuration.
+ * @property {string} callbackUri - The URI for the OAuth2 callback after authentication.
+ * @property {{ [key: string]: string }} [callbackUriParams] - Optional additional parameters for the callback URI.
+ */
+export interface BasicAuthConfig {
```

Comment:
I found that name in OAuth may be confused, actually BasicAuthConfig cannot store pw and username

Reply:
The pw and username are the service account username and secret; I thought that's the credential we're supposed to put in .env (like the clientId and client secret of OAuth)? The authenticate() method basically just verifies it, and the verify() method retrieves an expiration date (generally service accounts don't expire so it would be `null` most of the time) and some other info. We should discuss it tmr, I'm a bit confused lol